prog/vnet/load_balancer_nexus.rb

```diff
@@ -170,23 +173,12 @@ def need_to_rewrite_dns_records?
     load_balancer.dns_zone&.delete_record(record_name: load_balancer.hostname)
 
     load_balancer.vms_to_dns.each do |vm|
-      # Insert IPv4 record if stack is ipv4 or dual, and vm has IPv4
-      if load_balancer.ipv4_enabled? && vm.ip4_string
-        load_balancer.dns_zone&.insert_record(
-          record_name: load_balancer.hostname,
-          type: "A",
-          ttl: 10,
-          data: vm.ip4_string
-        )
-      end
-
-      # Insert IPv6 record if stack is ipv6 or dual
-      if load_balancer.ipv6_enabled?
+      vm_dns_records(vm).each do |type, data|
```

Comment:
Might be worth surrounding the `each` call inside `if load_balancer.dns_zone`, and drop the `&.`, so you only check it once and not twice. Up to you.